### PR TITLE
Crpogace/2.7.180 fixes

### PR DESCRIPTION
**Pull request by cristipogacean**
* fixed creation of duplicate dataSetWriterId in job definition bug
* changed NodePublishBulkAsync behavior to generate a new dataSetWriterId respective a new dataset for every new api call
* added unit tests for Publish Bulk API
* added host bind mounts for layered IoT Edge deployments
* PkiRootPath is set to the same mount storage for all edge modules so that the modules can share the same pki store

### CODE REVIEW

**Comment by karok2m** [comment]
Original pull requests for `release/2.7.180` branch:
* [Bugfix/publisher #689](https://github.com/Azure/Industrial-IoT/pull/689)
* [bind mounts for layered Iot Edge deployments #691](https://github.com/Azure/Industrial-IoT/pull/691)

**Review by barnstee — DISMISSED** [review]

**Comment by karok2m** [comment]
As per discussion with @cristipogacean, I am adding `in progress` label since he plans to add his other changes to `release/2.7.180` branch to this PR:
* [Hotfix/edge update #711](https://github.com/Azure/Industrial-IoT/pull/711)

**Comment by karok2m** [comment]
Another set of changes from `release/2.7.180` branch that is present in this PR:
* fix updated job with new iot device conn. string #726

**Review by karok2m — CHANGES_REQUESTED** [review]

File: common/src/Microsoft.Azure.IIoT.Core/src/Hub/Services/IoTHubEdgeBaseDeployment.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ public sealed class IoTHubEdgeBaseDeployment : IHostProcess {
                     ""restartPolicy"": ""always"",
                     ""settings"": {
                         ""image"": ""mcr.microsoft.com/azureiotedge-hub:" + version + @""",
-                        ""createOptions"":  ""{\""HostConfig\"":{\""PortBindings\"":{\""443/tcp\"":[{\""HostPort\"":\""443\""}],\""5671/tcp\"":[{\""HostPort\"":\""5671\""}],\""8883/tcp\"":[{\""HostPort\"":\""8883\""}],\""9600/tcp\"":[{\""HostPort\"":\""9600\""}]}},\""ExposedPorts\"":{\""5671/tcp\"":{},\""8883/tcp\"":{},\""9600/tcp\"":{}}}""
+                        ""createOptions"": ""{\""HostConfig\"":{\""PortBindings\"":{\""443/tcp\"":[{\""HostPort\"":\""443\""}],\""5671/tcp\"":[{\""HostPort\"":\""5671\""}],\""8883/tcp\"":[{\""HostPort\"":\""8883\""}]}},\""ExposedPorts\"":{\""5671/tcp\"":{},\""8883/tcp\"":{}}}""
```
[the remark this answers is not in the document]
Comment (cristipogacean) [reply common/src/Microsoft.Azure.IIoT.Core/src/Hub/Services/IoTHubEdgeBaseDeployment.cs]: @karok2m, thanks for bringing this up! this change was unintended, most likely the result of some conflicted or failed  master merge!? it's fixed now

**Review by karok2m — APPROVED** [review]

**Review by barnstee — APPROVED** [review]

**Pull request merged**